### ISSUE TITLE
[Enhancement] Add different input types for prompt 

### ISSUE THREAD

**Issue by angelru** [dotnet/maui#9913]
### Description

https://github.com/xamarin/Xamarin.Forms/issues/8363

We have now implemented a XamForms https://github.com/xamarin/Xamarin.Forms/issues/6713, and the ability to set the https://github.com/xamarin/Xamarin.Forms/issues/8346, meanwhile some people have asked for https://github.com/xamarin/Xamarin.Forms/pull/6714#issuecomment-544546925. I.e. password, but we can think of more like numeric or dates, etc.

Since an issue wasn't opened for it yet, I'm opening it to track it properly.

### Public API Changes

It probably makes sense to introduce an enum to be able to facilitate more different input types. Note that I'm just making up a list now as I go. It would probably be best to implement different input types one by one in different PRs to validate their use first.

```
public enum PromptInputType
{
    Text,
    Password,
    Numeric,
    Date,
    Time
   // etc.
}
```
Then we should pass that to the Page.DisplayPromptAsync that is available today.

`public Task<string> DisplayPromptAsync(string title, string message, PromptInputType inputType, string accept = "OK", string cancel = "Cancel", string placeholder = null, int maxLength = -1, Keyboard keyboard = default(Keyboard), string initialValue = "")`

Notice that I added the PromptInputType inputType, in there.

Now in the implemented platforms we need to implement the actual representations of these input types.

### Intended Use-Case

Have more specific prompts to the type of data that should be entered

**Comment by Cruik** [comment]
Any progess?

**Comment by angelru** [comment]
> Any progess?

I don't know

**Comment by jakob-bmd** [comment]
/bump
Would also very much like to see this feature!

**Comment by Jerome2606** [comment]
/bump 
Mandatory feature for a smooth migration

**Comment by pictos** [comment]
@rmarinho, since my PR was merged on net9 branch, this issue needs to be closed manually

**Comment by rmarinho** [comment]
Thanks, closing fixed by #20194
